src/rngs/os.rs

```diff
@@ -834,37 +877,31 @@ mod imp {
     #[derive(Clone, Debug)]
     pub struct OsRng;
 
-    impl OsRng {
-        pub fn new() -> Result<OsRng, Error> {
-            Ok(OsRng)
-        }
+    impl OsRngImpl for OsRng {
+        fn new() -> Result<OsRng, Error> { Ok(OsRng) }
 
-        pub fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error> {
-            // RtlGenRandom takes an ULONG (u32) for the length so we need to
-            // split up the buffer.
-            let max = <ULONG>::max_value() as usize;
-            if dest.len() <= max {
-                trace!("OsRng: reading {} bytes via RtlGenRandom", dest.len());
-            } else {
-                trace!("OsRng: reading {} bytes via RtlGenRandom in {} chunks of {} bytes",
-                       dest.len(), (dest.len() + max) / max, max);
-            }
-            for slice in dest.chunks_mut(max) {
-                let ret = unsafe {
-                    RtlGenRandom(slice.as_mut_ptr() as PVOID, slice.len() as ULONG)
-                };
-                if ret == 0 {
-                    return Err(Error::with_cause(
-                        ErrorKind::Unavailable,
-                        "couldn't generate random bytes",
-                        io::Error::last_os_error()));
-                }
+        fn fill_chunk(&mut self, dest: &mut [u8]) -> Result<(), Error> {
+            let ret = unsafe {
+                RtlGenRandom(dest.as_mut_ptr() as PVOID, dest.len() as ULONG)
+            };
+            if ret == 0 {
+                return Err(Error::with_cause(
+                    ErrorKind::Unavailable,
+                    "couldn't generate random bytes",
+                    io::Error::last_os_error()));
             }
             Ok(())
         }
+
+        fn max_chunk_size(&self) -> Option<usize> {
+            Some(<ULONG>::max_value() as usize)
```

Comment:
I assume this is `u32::MAX`. Should probably be the default; it's not like anyone will use more anyway.